application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java

```diff
@@ -140,6 +141,7 @@ public void updateDesignatedBodyCode(final ConnectionMessageDto message) {
       final var doctorsForDB = doctorsForDBOptional.get();
       doctorsForDB.setDesignatedBodyCode(dbc);
       doctorsForDB.setExistsInGmc(dbc != null);
+      if(dbc == null) doctorsForDB.setUnderNotice(NO);
```

Comment:
We evaluate whether DBC is null twice.  I would guess that the compiler isn't smart enough to reduce the number of evaluations here... It's worth thinking about the trade off in increased LOC & additional processing.

It's a tiny difference at this scale but we should try and do the optimal thing.